src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCSVFileRecord.java

```diff
@@ -516,4 +522,62 @@ public boolean next() throws SQLServerException {
         }
         return (null != currentLine);
     }
+
+    /**
+     * Returns whether the rules to escape delimiters are used.
+     *
+     * @return true if the rules are used, false otherwise.
+     */
+    public boolean isEscapeColumnDelimitersCSV() {
+        return escapeDelimiters;
+    }
+
+    /**
+     * When set to true, the following rules will be used to parse CSV files: Each field may or may not be enclosed in
+     * double quotes. If fields are not enclosed with double quotes, then double quotes may not appear inside the
+     * fields. Fields containing double quotes, and delimiters should be enclosed in double quotes. If double-quotes are
+     * used to enclose fields, then a double-quote appearing inside a field must be escaped by preceding it with another
+     * double quote. Spaces are considered part of a field. Spaces before and after enclosing double quotes are ignored.
+     *
+     * @param escapeDelimiters
+     *        true if the rules above to be used.
+     */
+    public void setEscapeColumnDelimitersCSV(boolean escapeDelimiters) {
+        this.escapeDelimiters = escapeDelimiters;
+    }
+
+    private static String[] escapeQuotesRFC4180(String[] tokens) throws SQLServerException {
+        if (null == tokens) {
+            return tokens;
+        }
+        for (int i = 0; i < tokens.length; i++) {
+            boolean escaped = false;
+            int j = 0;
+            StringBuilder sb = new StringBuilder();
+            long quoteCount = tokens[i].chars().filter(ch -> ch == '"').count();
```

Comment:
This lambda expression would slow down the parsing of data by scanning through every character. The if statement on line 569 already implicitly does this check, where adding an else statement to

```if ((j < tokens[i].length() - 1) && '"' == tokens[i].charAt(j + 1)) {```

this if statement would guarantee that the corrupt String would be caught.

Reply:
Actually, adding an else statement wouldn't catch all the cases; if the String starts with a double quote but doesn't end in one, or if the String starts with a double quote but ends with an escaped double quote would not fail in the else statement. If we want to make it easy, we can just increment a counter to count the number of double quotes as we go along in the while loop, instead of counting it here.

Reply:
I improved the error handling a bit, but kept the quoteCount part as it is. `filter()` has the same time complexity as `contains()` and I need to use one of them to tell if the driver is parsing a double quote enclosed field. I looked into counting double quotes in the while loop, but it becomes significantly challenging to ignore the whitespaces after the last quote. I think it is okay to keep `filter()` because the time complexity still remains O(n).

Reply:
Thanks, that sounds good to me.